docs/guides/server/windows-service.adoc

```diff
@@ -0,0 +1,174 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+    title="Run {project_name} as a Windows Service"
+    summary="Install and run {project_name} as a Windows service using Apache Commons Daemon.">
+
+This guide explains how to install and run {project_name} as a Windows service using Apache Commons Daemon. The service runs `kc.bat` in "exe" mode so behavior matches running `kc.bat start` manually. The service runs in "exe" mode, where Procrun executes `kc.bat start` as an external process. Environment variables, such as `KC_*`, along with `conf/keycloak.conf`, are respected. The `kc.bat` script handles augmentation and build logic, ensuring the service behaves exactly like a manual start.
+
+==  Apache Commons Daemon Setup
+To run {project_name} as a Windows service, you need the Apache Commons Daemon Procrun binary (`prunsrv.exe`). Download it for your platform: https://downloads.apache.org/commons/daemon/binaries/windows/
+
+Then place `prunsrv.exe` into the Keycloak `bin` folder.
+
+[source,bash]
+----
+copy "path\to\prunsrv.exe" "%KEYCLOAK_HOME%\bin\prunsrv.exe"
+----
+
+Use the amd64 binary for 64-bit and x86 for 32-bit systems.
+
+== Optional: Pre-build
+
+Pre-building is optional. If you do not pre-build, Keycloak will build automatically on first start, which takes longer.
+
+[source,bash]
+----
+bin/kc.bat build --db=postgres
+----
+
+== Creating the Service
+
+Keycloak includes a `tools win-service` subcommand to simplify service creation and deletion.
+
+[source,bash]
+----
+bin\kc.bat tools win-service create --help
+----
+
+[source,bash]
+----
+bin\kc.bat tools win-service delete --help
+----
+
+=== Examples
+
+Create a basic service (runs as Local System by default):
+
+[source,bash]
+----
+bin\kc.bat tools win-service create --name keycloak
+----
+
+Manual startup and longer stop timeout:
+
+[source,bash]
+----
+bin\kc.bat tools win-service create --startup=manual --stop-timeout=60
+----
+
+Delayed auto-start:
+
+[source,bash]
+----
+bin\kc.bat tools win-service create --startup=delayed
```

Comment:
The current warning for unnecessary cli options is limited to runtime variables only. However for the tools commands that is not correct.

If you do something like: `kc.bat tools win-service create --startup=delayed --db=oracle`

There won't be a warning that --db=oracle is not used.

We should either handle that warning or for commands that don't accept either buildtime or runtime options, just error when unrecognized args are provided.

This of course can be done in a follow-up.

Reply:
@shawkins if I read the code the right way, it happens because it checks if PropertyMappers.getMapperByCliKey("--db") returns a mapper and since `--db` is a valid option, it is removed from unrecognized args?

In that case should I add something like this to Picocli?
<img width="2342" height="1101" alt="image" src="https://github.com/user-attachments/assets/5c52be21-42e6-44ea-959b-90b765104fad" />
This way such options should not be removed from unrecognized args. But just an idea, I haven't tried it yet.

Anyway, it probably touches more that I would want to, at this time of the year :). So you are right with the follow-up.